lib/AST/Type.cpp

```diff
@@ -3703,8 +3703,9 @@ PackArchetypeType::PackArchetypeType(
     ArrayRef<ProtocolDecl *> ConformsTo, Type Superclass,
     LayoutConstraint Layout, PackShape Shape)
     : ArchetypeType(TypeKind::PackArchetype, Ctx,
-                    RecursiveTypeProperties::HasArchetype, InterfaceType,
-                    ConformsTo, Superclass, Layout, GenericEnv) {
+                    RecursiveTypeProperties::HasArchetype |
+                        RecursiveTypeProperties::HasConcretePack,
```

Comment:
It’s actually PackExpansionType that leads us into this situation

Reply:
Yeah, it's a PackExpansionType whose pattern is a PackArchetypeType:

```
(tuple_type num_elements=1
  (tuple_type_elt
    (pack_expansion_type
      (pattern=pack_archetype_type address=0x127045b10
        (interface_type=dependent_member_type assoc_type=Swift.(file).Sequence.Element
          (base=generic_type_param_type depth=0 index=0 decl=main.(file).tupleExpansionWithMemberType(seqs:elts:).T@main.swift:1:47 pack)) name=(each T).Element
)
      (count=pack_archetype_type address=0x127045880
        (interface_type=generic_type_param_type depth=0 index=0 decl=main.(file).tupleExpansionWithMemberType(seqs:elts:).T@main.swift:1:47 pack) conforms_to=Swift.(file).Sequence name=each T
))))
```